rust/src/http2/http2.rs

```diff
@@ -1144,6 +1153,30 @@ impl HTTP2State {
                         return AppLayerResult::err();
                     }
                     let tx = tx.unwrap();
+                    let _pdu = Frame::new(
```

Comment:
why do frames depend on a tx being created? In what case would this fail?

Reply:
> In what case would this fail?

In case there are too many live transactions

> why do frames depend on a tx being created?

Ok, next version creates the frame first and sets the tx id if we are in the regular case a few lines later